mlflow/store/tracking/utils/sql_trace_metrics_utils.py

```diff
@@ -123,13 +124,109 @@ def get_time_bucket_expression(
         return func.floor(timestamp_column / bucket_size_ms) * bucket_size_ms
 
 
+def _parse_trace_metrics_filter(filter_string: str) -> list[dict[str, Any]]:
+    """
+    Parse filter string with 'trace.' prefix for trace metrics.
+
+    Args:
+        filter_string: Filter in format 'trace.xxx = value'
+
+    Returns:
+        List of parsed filter clauses
+    """
+    stripped = filter_string.strip()
+    if not stripped.startswith("trace."):
+        raise MlflowException.invalid_parameter_value(
+            f"Filter must start with 'trace.' prefix, got: '{filter_string}'"
+        )
+
+    # Remove 'trace.' prefix and parse with SearchTraceUtils
+    filter_without_prefix = stripped.removeprefix("trace.")
+    return SearchTraceUtils.parse_search_filter_for_search_traces(filter_without_prefix)
+
+
+def apply_global_filters(query: Query, filters: list[str] | None) -> Query:
+    """
+    Apply global filters to the trace query.
+
+    Supported filters:
+    - trace.status = 'OK': Filter by trace status
+    - trace.metadata.`mlflow.sourceRun` = 'run_123': Filter by metadata
+    - trace.tag.`model.version` = 'v1': Filter by tag value
+
+    All filters in the list are combined with AND logic.
+
+    Args:
+        query: SQLAlchemy query to filter
+        filters: List of filter strings, each in format 'trace.xxx = value'
+
+    Returns:
+        Filtered query
+
+    Note:
+        Only '=' operator is supported. All filters must start with 'trace.' prefix.
+    """
+    if not filters:
+        return query
+
+    for filter_string in filters:
+        parsed_filters = _parse_trace_metrics_filter(filter_string)
+
+        for filter_clause in parsed_filters:
+            type_ = filter_clause.get("type")
+            key = filter_clause.get("key")
+            value = filter_clause.get("value")
+            comparator = filter_clause.get("comparator", "").upper()
```

Comment:
Do we need the upper() command and the empty string fallback if we're immediately asserting equivalency to `"="`?